kubernetes-kit-starter/src/main/java/com/vaadin/kubernetes/starter/sessiontracker/UnserializableComponentWrapper.java

```diff
@@ -136,6 +144,7 @@ private void restoreComponent() {
                     getElement().appendChild(component.getElement());
                 }
             } finally {
+                CurrentInstance.restoreInstances(instances);
```

Comment:
We should call `CurrentInstance.clearAll()` before `CurrentInstance.restoreInstances(...)` to prevent leaking instances outside this method.

Note: the same should be done in `TransientAwareHolder`